src/node_buffer.cc

```diff
@@ -340,7 +344,9 @@ void Copy(const FunctionCallbackInfo<Value> &args) {
 
 
 void Fill(const FunctionCallbackInfo<Value>& args) {
-  ARGS_THIS(args[0].As<Object>())
+  Environment* env = Environment::GetCurrent(args);
```

Comment:
Ah wait. I see. It's because you're throwing from `ARGS_THIS()`. Missed that with the load of comments around the code.